cmake: update to cmake: 3.15.7

ioannisg opened the pull request:
Move cmake version to 3.15.7.
This is required for building Zephyr with TF-M support.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
cmake: update to cmake: 3.15.7

Move cmake version to 3.15.7.
Building Zephyr with TF-M support requires 3.15.0,
so we update cmake to the latest stable in the
3.15.x series.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg commented:
cc @galak @carlescufi @nashif

galak commented on shippable/install_source.sh:
```diff
@@ -11,7 +11,7 @@ cd ..
 
 
-CMAKE_VERSION=3.13.1
+CMAKE_VERSION=3.15.0
```
Probably best for this to be 3.15.6 ( I think that is the latest dot release )

carlescufi replied:
Agreed, we should use the latest stable in the series.

ioannisg replied:
Yeap, sure, I'll update this.

carlescufi replied:
Latest Ubuntu LTS comes with 3.16.3: https://packages.ubuntu.com/focal/cmake, so I think 3.15.6 is a good choice.

galak replied:
Looks like v3.15.7 based on the git tags for CMake repo.

nashif replied:
just make sure it works with our current tree and does not generate all kind of policy warnings that would force us to move minimum required version before we know it....

ioannisg replied:
Yeap, 3.15.7 exists in GH

ioannisg committed:
python: add imgtool in the packages to be installed

Add imgtool in the packages to be installed.
imgtool is needed when building Zephyr with TF-M.
It is currently not covered by MCUboot requirements.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

galak commented on shippable/install.sh:
```diff
@@ -145,3 +145,9 @@ pip3 install -q statistics numpy
 pip3 install -r https://raw.githubusercontent.com/JuulLabs-OSS/mcuboot/master/scripts/requirements.txt
 
+# imgtool, needed for MCUboot to sign images (i.e. for TF-M S/NS builds)
+pip3 install -q imgtool
```
Not related to this PR, but do we know why imgtool isn't in https://raw.githubusercontent.com/JuulLabs-OSS/mcuboot/master/scripts/requirements.txt ?

ioannisg replied:
Good point, I was actually looking into the mcu boot requirements, and plan to check with David.

ioannisg replied:
@galak it is now: https://github.com/mcu-tools/mcuboot/blob/master/scripts/requirements.txt

ioannisg replied:
though the contents are the same :)

galak replied:
do you mind adding a commit to move to ` https://github.com/mcu-tools`

ioannisg replied:
yeap, sure, done

ioannisg commented:
Extra commit is for adding imgtool and lpc_checksum python packages, which is currently done in .buildkite as a temp-hack.

ioannisg committed:
python: update link to mcuboot requirements.txt

Update Github link to the mcuboot requirements.txt.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

galak commented on shippable/install.sh:
```diff
@@ -143,5 +143,11 @@ pip3 install -q statistics numpy
+pip3 install -q imgtool
+
+# lpc_checksum, needed to build images for lpcxpresso11u68 boards
+pip3 install -q lpc_checksum
```
We can remove this, it should be covered by the zephyr requirements.txt

ioannisg replied:
> We can remove this, it should be covered by the zephyr requirements.txt

Just to clarify @galak: why don't we do the same for imgtool?

One option would be to keep it here temporarily until it is passed in mcuboot requirements.txt.
But then, imgtool is used when we integrate with TFM (which uses it's own mcuboot, so then it "becomes" a requirement for zephyr)

ioannisg replied:
For now I kept imgtool, as it is not covered by neither mcuboot nor zephyr requirements.txt.

carlescufi replied:
I agree, why not add it to `requirements-extras.txt`? I mean, we are already installing it in CI via hack:
https://github.com/zephyrproject-rtos/zephyr/blob/master/.buildkite/run.sh#L58

carlescufi replied:
>Just to clarify @galak: why don't we do the same for imgtool?

`imgtool` is not in our requirements.txt today.

ioannisg replied:
> imgtool is not in our requirements.txt today.
@carlescufi i meant, why don't we add it in our requirements, as well?

galak replied:
@carlescufi my comment was w/regards to lpc_checksum being in requirements.txt (specifically requirements-extras.txt)

ioannisg commented:
Does this repo automatically dismiss reviews on pushing updates?

galak commented:
> Does this repo automatically dismiss reviews on pushing updates?

Not sure.  Maybe some new github feature.

galak commented:
> Not sure. Maybe some new github feature.

looks like we have `Dismiss stale pull request approvals when new commits are pushed` set.
